source/exe/hot_restart.cc

```diff
@@ -307,7 +308,17 @@ void HotRestartImpl::sendMessage(sockaddr_un& address, RpcBase& rpc) {
 
 void HotRestartImpl::onGetListenSocket(RpcGetListenSocketRequest& rpc) {
   RpcGetListenSocketReply reply;
-  reply.fd_ = server_->getListenSocketFd(std::string(rpc.address_));
+  reply.fd_ = -1;
+
+  Network::Address::InstanceConstSharedPtr addr =
+      Network::Utility::resolveUrl(std::string(rpc.address_));
+  for (const auto& listener : server_->listenerManager().listeners()) {
+    if (listener.get().socket().localAddress()->asString() == addr->asString()) {
+      reply.fd_ = listener.get().socket().fd();
```

Comment:
Do we not have equality defined on Address::Instance? I think it's better generally to be using these logical equality operations rather than degenerating into string representations.